app/grandchallenge/cases/views.py

```diff
@@ -139,3 +139,33 @@ def get_context_data(self, **kwargs):
             }
         )
         return context
+
+
+class CSImageDetail(
+    LoginRequiredMixin, ObjectPermissionRequiredMixin, DetailView
+):
+    model = Image
+    permission_required = (
+        f"{Image._meta.app_label}.view_{Image._meta.model_name}"
+    )
+    raise_exception = True
+    login_url = reverse_lazy("account_login")
+    template_name = "cases/image_detail_cs.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        try:
+            self.object.get_metaimage_files()
+        except FileNotFoundError as e:
+            raise Http404 from e
+
+        if self.object.color_space not in (
+            Image.COLOR_SPACE_GRAY,
+            Image.COLOR_SPACE_RGBA,
+            Image.COLOR_SPACE_RGB,
+        ):
+            # YCrBr not supported in cornerstone
+            raise Http404
+
+        context.update({"image_pk": self.object.pk})
```

Comment:
I don't think that you need to set this, you can use use `object.pk` in the template directly.